### PR TITLE
Schedule class jazz

### PR DESCRIPTION
**Pull request by jengu288**
you know what it is

### CODE REVIEW

File: server/schedule.py [hunk 1]
```diff
@@ -1,19 +1,49 @@
...
+
+        if uid in ids:
+            self.mem_sched[timestamp].remove(uid)
+            self._update_date(timestamp)
 
     def get(self, date) -> list:
-        pass
+        timestamp = datetime.timestamp(date)
+
+        try:
```
Comment (ConorMurphy21) [inline server/schedule.py:32]: Make this a function?

File: server/schedule.py [hunk 2]
```diff
@@ -1,19 +1,49 @@
...
+            ids = self.mem_sched[timestamp]
+        except KeyError:
+            ids = self.mem_sched[timestamp] = self._read_file(timestamp)
+        return ids
 
     def get_week(self, date) -> list:
-        pass
+        dt_object = datetime.fromtimestamp(date)
```
Comment (ConorMurphy21) [inline server/schedule.py:39]: Add a todo or fixme for this function, so we know it's broken

File: server/schedule.py [hunk 3]
```diff
@@ -23,9 +53,23 @@ def _store(self):
...
+    def _append_to_date(self, uid, timestamp):
+        f = open(self.directory + str(timestamp) + ".txt", "a")
+        f.write(uid + '\n')
+        f.close()
+
+    def _read_file(self, timestamp) -> list:
+        # load in from disc
+        filename = self.directory + str(timestamp) + ".txt"
+        # create file if doesn't exist (a+)
+        with open(filename, "a+") as f:
```
Comment (ConorMurphy21) [inline server/schedule.py]: a+ is for appending, but you then seek(0), I think r+ will seek(0) by default

**Review by ConorMurphy21 — CHANGES_REQUESTED** [review]

File: server/schedule.py (hunk 3 above)
Comment (jengu288) [reply server/schedule.py]: @ConorMurphy21 i've heard that r+ doesnt create a file if it doesn't exist. a lot of the research I did didn't really clarify. can you briefly look into it?

Comment (ConorMurphy21) [reply server/schedule.py]: @jengu288  you are reading anyways, if it doesn't create a file that means there is nothing on disk. So just set your return value above the with to be empty?

**Pull request merged**